spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -112,6 +113,43 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     toDataset(sparkSession, restore)
   }
 
+  protected def executeClone(
+    table: DeltaTableV2,
+    target: String,
+    create: Boolean,
+    replace: Boolean,
+    ifNotExists: Boolean,
+    tableProperties: Map[String, String],
+    versionAsOf: Option[Long] = None,
+    timestampAsOf: Option[String] = None
+  ): DataFrame = withActiveSession(sparkSession) {
```

Comment:
I think `executeClone` should return `DeltaTable`, so that calling `clone` on a source `DeltaTable` returns a target `DeltaTable`

Reply:
You would lose the stats that get returned in the DataFrame if you returned the DeltaTable though

Reply:
@Kimahriman Hmm, how do you know this? Using `io.delta.tables.DeltaTable.forPath` or `io.delta.tables.DeltaTable.forName` lose stats?

Reply:
I guess that's how creating a table works so probably makes sense, updated

Reply:
I just mean it was currently returning a DataFrame that has stats from the clone operation. If you return the DeltaTable instead you don't get those stats anymore returned.